ote_sdk/ote_sdk/tests/entities/test_resultset.py

```diff
@@ -0,0 +1,126 @@
+from ote_sdk.tests.constants.ote_sdk_components import OteSdkComponent
+from ote_sdk.tests.constants.requirements import Requirements
+from ote_sdk.utils.time_utils import now
+
+
+@pytest.mark.components(OteSdkComponent.OTE_SDK)
+class TestResultset:
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_resultset_purpose(self):
+        """
+        <b>Description:</b>
+        Check the ResultsetPurpose can correctly return the value
+
+        <b>Input data:</b>
+        Denotes, stages
+
+        <b>Expected results:</b>
+        Test passes if the results match
+        """
+
+        denotes = ["EVALUATION", "TEST", "PREEVALUATION"]
+        stages = ["Validation", "Test", "Pre-validation"]
+
+        resultset_purpose = ResultsetPurpose
+        assert len(resultset_purpose) == 3
+
+        for i in ResultsetPurpose:
+            resultset_purpose = ResultsetPurpose(i)
+            assert repr(resultset_purpose) == denotes[i.value]
+            assert str(resultset_purpose) == stages[i.value]
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_resultset_entity(self):
+        """
+        <b>Description:</b>
+        Check the ResultSetEntity can correctly return the value
+
+        <b>Input data:</b>
+        Mock data
+
+        <b>Expected results:</b>
+        Test passes if incoming data is processed correctly
+
+        <b>Steps</b>
+        1. Create dummy data
+        2. Check the processing of default values
+        3. Check the processing of changed values
+        """
+
+        test_data = {
+            "model": None,
+            "ground_truth_dataset": None,
+            "prediction_dataset": None,
+            "purpose": None,
+            "performance": None,
+            "creation_date": None,
+            "id": None,
+        }
+
+        result_set = ResultSetEntity(**test_data)
+
+        for name, value in test_data.items():
+            set_attr_name = f"test_{name}"
+            if name in [
+                "model",
+                "ground_truth_dataset",
+                "prediction_dataset",
+                "purpose",
+            ]:
+                assert getattr(result_set, name) == value
+                setattr(result_set, name, set_attr_name)
+                assert getattr(result_set, name) == set_attr_name
+
+        assert result_set.performance == NullPerformance()
+        assert type(result_set.creation_date) == datetime
+        assert result_set.id == ID()
+
+        assert result_set.has_score_metric() is False
+        result_set.performance = "test_performance"
+        assert result_set.performance != NullPerformance()
+        assert result_set.has_score_metric() is True
+
+        result_set.creation_date = now().replace(microsecond=0)
+        assert result_set.creation_date == now().replace(microsecond=0)
```

Comment:
I guess the issue still exists

Reply:
The error was on line 99, but even here it was replaced by a specific time.